flutterproject/flutterproject.go

```diff
@@ -0,0 +1,50 @@
+package flutterproject
+
+type SDKVersionsReader interface {
+	ReadSDKVersions(projectRootDir string) (*VersionConstraint, *VersionConstraint, error)
+}
+
+type Project struct {
+	rootDir    string
+	fileOpener FileOpener
+}
+
+func New(rootDir string, fileOpener FileOpener) Project {
+	return Project{
+		rootDir:    rootDir,
+		fileOpener: fileOpener,
+	}
+}
+
+type FlutterAndDartSDKVersions struct {
```

Comment:
FlutterAndDartSDKVersions should store the version constraint sources as dedicated fields.